Code/Framework/AzCore/AzCore/Serialization/Json/JsonImporter.cpp

```diff
@@ -144,21 +160,606 @@ namespace AZ
+            {
+                targetFieldValue->CopyFrom(fieldValue, allocator, true);
+            }
+            else
+            {
+                if (source.IsObject())
+                {
+                    rapidjson::Value name;
+                    rapidjson::Value value;
+                    name.CopyFrom(rapidjson::Value(fieldNameStringRef), allocator, true);
+                    value.CopyFrom(fieldValue, allocator, true);
+                    target.AddMember(AZStd::move(name), AZStd::move(value), allocator);
+                }
+                else
+                {
+                    // Push the value to the end of the array
+                    rapidjson::Value value;
+                    value.CopyFrom(fieldValue, allocator, true);
+                    target.PushBack(AZStd::move(value), allocator);
+                }
+
+            }
+        }
+
+        return mergePatchResult;
+    }
+
+    JsonSerializationResult::ResultCode JsonImportResolver::ResolveImportsInOrder(rapidjson::Value& target,
+        rapidjson::Document::AllocatorType& allocator, const rapidjson::Value& source,
+        ImportPathStack& importPathStack,
+        JsonImportSettings& settings, StackedString& element)
+    {
+        using namespace JsonSerializationResult;
+
+        using ReporterString = AZStd::fixed_string<1024>;
+
+        if (settings.m_importer == nullptr)
+        {
+            ReporterString reportMessage = "Json Importer is nullptr for ResolveImportsInOrder.";
+            if (!importPathStack.empty())
+            {
+                reportMessage += ReporterString::format(R"( Cannot resolve imports in field order for file "%s".)", importPathStack.back().c_str());
+            }
+            return settings.m_reporting(reportMessage,
+                ResultCode(Tasks::Import, Outcomes::Skipped), element);
+        }
+
+        // If the source json is neither and Object or an Array
+        // Just make a copy of the data and return
+        if (!source.IsObject() && !source.IsArray())
+        {
+            target.CopyFrom(source, allocator, true);
+            return ResultCode(Tasks::Import, Outcomes::Success);
+        }
+
+        ResultCode resolveImportsResult(Tasks::Import, Outcomes::Success);
+
+        // Stores the currently copied field from the source object or array
+        // into a local JSON value that will be moved to the target JSON value at the end of this function
+        rapidjson::Value targetDataAtCurrentDepth{ source.IsObject()
+            ? rapidjson::kObjectType : rapidjson::kArrayType };
+
+        // If this point is reached, the source is either an object or an array
+        size_t fieldCount = source.IsObject() ? source.MemberCount() : source.Size();
+        for (size_t fieldIndex = 0; fieldIndex < fieldCount; ++fieldIndex)
+        {
+            // Used to store the converted field name for an array field
+            using FieldNameString = AZStd::fixed_string<256>;
```

Comment:
If this is an expected limitation of field name sizes, the 256 should get promoted somewhere more visible.

Reply:
The FieldNameString is just used to store a converted array index to into a string value.

The Field length is only limited by the size of rapidjson.
That being said the FieldNameString can be a lot smaller, since only 19 characters are needed to represent index values up to 2^64.

Reply:
Ah, thanks! I'd suggest that you add that information into the comment above it, I didn't realize it was holding an array index.
```
// Used to store the converted array index for an array field. Since it holds a converted 64-bit number, it will never be larger than 19 characters.
```

Reply:
I am updating it to calculate the exact number of digits needed to store the maximum value of the rapidjson size type rounded up to the nearest power of 2.